include/envoy/buffer/buffer.h

```diff
@@ -55,6 +55,25 @@ class BufferFragment {
   virtual void done() PURE;
 };
 
+/**
+ * A wrapper class to facilitate extracting buffer slices from a buffer instance.
+ */
+class SliceData {
+public:
+  virtual ~SliceData() = default;
+  /**
+   * @return void* a pointer to the data.
+   */
+  virtual void* data() PURE;
```

Comment:
Suggestion:  replace the `data` and `size` methods with:
`virtual absl::Span<uint8_t> getData() PURE;`
.. which would also simplify using it as a base class instead of wrapper.

Reply:
Suggestion implemented as part of commit 4d54587